docs/FirstUsage.md

```diff
@@ -0,0 +1,148 @@
+{
+
+
+    $Configuration = 'Debug'
+    $Environment = 'Development'
+
+
+}
+```
+
+We can start by creating simple task in `scripts\BuildTasks.ps1` file. Add following content to the end of file:
+
+```powershell
+Task test `
+    -description 'Test task.' `
+    -requiredVariables @('Param') `
+{
+    Write-Information("You are using configuration $Configuration with param value $Param.");
+}
+```
+
+It is recommended to have all parameters required for tasks in properties section. So now to use `Param` variable you should update properties section:
+
+```powershell
+Properties `
+{
+    $Param = $null
+}
+```
+
+With this `Param` variable will not be defined even if you provide it within command line. Now you can call it:
+
+```shell
+psake test -properties @{Param=123}
+```
+
+The output should be:
+
+```
+You are using configuration Debug with param value 123.
+```
+
+We have setup simple task with one required parameter `param`. In tasks global properties are available for you as parameters. For example `$Configuration` is reserved for solution build configuration and `$Environment` is for current used environment. By setting different values to script we can change task behavior. In current example we need to provide value for `Param` and have to pass `-properties` with Psake call that would override default value:
+
+```shell
+psake test -properties @{Configuration='Release';Param=123}
+```
+
+Now it is done! You can type `psake -docs` and see that our test task is enumerated. We can go ahead a do something useful.
+
+## Build and Publish Project
+
+The idea of PSGallery project is to provide set of useful Powershell functions to make CI process easier. Let's build and publish our project to remote Windows server! To build use `Invoke-SolutionBuild` function.
+
+**Note** The server should be configured and IIS needs to be installed. You should have remote management service running (see [WinRM Configuration](WinRMConfiguration.md) section) and IIS with WebDeploy installed. You can do this by run `Install-Iis $ServerHost -ManagementService -WebDeploy` on your local host.
+
+```powershell
+Task build `
+    -depends pre-build `
+    -description '* Build project.' `
+    -requiredVariables @('Configuration') `
+{
+    Initialize-MSBuild
+    Invoke-NugetRestore -SolutionPath "$src\WebApplication1.sln"
+    Invoke-SolutionBuild -SolutionPath "$src\WebApplication1.sln" -Configuration $Configuration
+}
+```
+
+You can run `psake build` to build project now. Some things to note:
+
+1. `-depends` keyword means that there are depended tasks need to be performed before.
+2. `$src` is already presented as global variable and means root directory for our source code.
+3. Even configuration variable is required there is a property `Configuration` and it will be used instead if no property passed.
+
+After build we try to deploy project to IIS server, for that use `Invoke-WebDeployment` function:
+
+```powershell
+Task deploy `
+    -depends build `
```

Comment:
Build task is redundant. Need to introduce `pre-build` and move NuGet restore there.